File: ruby/lib/agentic_commerce_protocol/models/risk_signal_type.rb
# frozen_string_literal: true
# This file is @generated
module AgenticCommerceProtocol
  class RiskSignalType
    CARD_TESTING = "card_testing".freeze

    def self.all_vars
      @all_vars ||= [CARD_TESTING].freeze
    end

    def initialize(value)
      unless RiskSignalType.all_vars.include?(value)
        raise "Invalid ENUM value '#{value}' for class #RiskSignalType"
      end

      @value = value
    end

    def self.deserialize(value)
      return value if RiskSignalType.all_vars.include?(value)
      raise "Invalid ENUM value '#{value}' for class #RiskSignalType"
    end

    def serialize
      @value
    end
  end
end

File: rust/src/models/risk_signal_type.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
pub enum RiskSignalType {
    #[serde(rename = "card_testing")]
    CardTesting,
}

impl fmt::Display for RiskSignalType {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let value = match self {
            Self::CardTesting => "card_testing",
        };
        f.write_str(value)
    }
}
